packages/diagrams/lib/src/gesture_detector.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:diagram_capture/diagram_capture.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'diagram_step.dart';
+
+const String _gestureDetector = 'gesture_detector';
+const Duration _pauseDuration = Duration(seconds: 1);
+final Duration _totalDuration = _pauseDuration + _pauseDuration;
+final GlobalKey _gestureDetectorKey = GlobalKey();
+
+class GestureDetectorDiagram extends StatefulWidget implements DiagramMetadata {
```

Comment:
This is another tricky one to visualize, I like it! What do you think about adding some padding and colorful icons to show it off a bit more? An ink splash might also help indicate that the text was pressed if you don't think that distracts too much from what you are trying to show. :)

Reply:
Yep, I can definitely make this one a bit prettier! 

I certainly agree an ink splash would also help folks understand the button is being pressed in the demo, but for that to work -- please correct me if I'm wrong! -- I believe we'd need to switch from using a GestureDetector to an InkWell instead, which might not be exactly what we want for these docs.

Reply:
All right -- I've added some padding to make it look a bit less busted and added a Lightbulb icon that Goes from Black -> Yellow when the button is tapped! Please let me know whatcha think